be/src/util/metrics.h

```diff
@@ -383,6 +383,15 @@ class MetricRegistry {
     mutable SpinLock _lock;
     std::map<std::string, MetricCollector*> _collectors;
     std::map<std::string, std::function<void()>> _hooks;
+
+    using FlatCollectors = std::vector<std::pair<std::string, MetricCollector>>;
```

Comment:
Why name flat?

Reply:
Use std::vector<std::pair> instead of std::map<k,v> so it is called flat collectors